react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: 0
+    };
+
+    this.timer = setInterval(() => {
+      if (this.state.index === this.props.pokemonList.length - 1) {
+        this.setState({ index: 0 });
+      } else {
+        this.setState({ index: this.state.index + 1 });
+      }
+    }, 3000);
```

Comment:
This is okay here, but it would be better to do this within the `componentDidMount` lifecycle method